dataset/gedi_rasterization.py

```diff
@@ -0,0 +1,103 @@
+
+import ee
+import json
+import os
+import datetime
+import time
+from ee.batch import Export
+import ast
+
+def make_timestampFromYMD(year, month, day):
+    tdat = datetime.datetime(year, month, day)
+    return time.mktime(tdat.timetuple()) * 1000
+
+def rasterize_GEDI_by_utm_zone(orbits, grid_id, month):
+    props = ["orbit", "track", "beam", "channel", "degrade", "dtime", "quality",
+             "rx_algrunflag", "rx_quality", "sensitivity", "toploc", "zcross",
+             "rh10", "rh20", "rh30", "rh40", "rh50", "rh60", "rh70", "rh80", "rh90", "rh98"]
+
+    def updateOrbit(f):
+        return (f.set('orbit', fshots.get('orbit'))
+                 .set('track', fshots.get('track'))
+                )
+
+    shots = []
+    for orbit in orbits:
+        fshots = ee.FeatureCollection(orbit).filterMetadata('quality', 'equals', 1)
+        augmented = fshots.map(updateOrbit)
+        shots.append(augmented)
+
+    shots = ee.FeatureCollection(shots).flatten()
+
+    grid = (ee.FeatureCollection("users/yang/GEETables/GEDI/GEDI_UTM_GRIDS_LandOnly")
```

Comment:
Could you point at the source URL of the UTM grids file?

Reply:
https://www.arcgis.com/home/item.html?id=b294795270aa4fb3bd25286bf09edc51